### PR TITLE
move psutil and pycurl out of base

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Notes:
- [x] exploring ways to get to interactive testing of the stack on binder python3.11 sooner rather than later
  - https://gist.github.com/bollwyvl/5f3efbca22eaafdd07ca2c87a6ac6e31
- [x] add `pycurl` and `psutil` to the `jupyterhub` metapackage
  - it doesn't appear these benefit the `single-user` experience, but almost certainly do the full spawner+X setup
- [x] add dedicated windows build with hard dep on `psutil`
- [x] add `__win` and `__unix` metapackages with `linux64` and `win64` builds 
- [x] make `pamela` only on `__unix` 
  - can't be removed and keep `pip check`
- [x] move `psutil >=5.6.5` to run-constrained, required on `win`
- [x] skip `configurable-http-proxy` check on windows
  -  might need investigation over there, e.g. `%SCRIPTS%\configurable-http-proxy.cmd`
- [x] add (failing) build artifact uploading in azure 
- [x] clean up, sort `conda-forge.yml`

### CODE REVIEW

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

**Comment by bollwyvl** [comment]
Oh, i see `psutil` is [conditionally required on windows](https://github.com/jupyterhub/jupyterhub/blob/f6eec29aa222276ea8a9d8066eabb53e9f65b1a7/requirements.txt#L11):

```
psutil>=5.6.5; sys_platform == 'win32'
```

**Comment by manics** [comment]
Looks good to me! `psutil` is only ultimately used in `app.py`, `pycurl` is to improve performance in the server so not needed in singleuser.

If you want to pare things down further you could probably also drop `pamela` since it's wrapped in a check:
https://github.com/jupyterhub/jupyterhub/blob/2594a7269e443f1856b825929fdc095d73ff6f5d/jupyterhub/auth.py#L15-L19

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

**Comment by bollwyvl** [comment]
> If you want to pare things down further you could probably also move `pamela`

Don't think we can, and still keep `pip check`...

**Comment by bollwyvl** [comment]
gah, forgot the [virtual packages](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-virtual.html#managing-virtual-packages).... want to see if win finishes, first, anyway....

**Comment by bollwyvl** [comment]
Seems legit to me, @conda-forge/jupyterhub ready for review

File: recipe/meta.yaml [hunk 1]
```diff
@@ -32,24 +32,29 @@ outputs:
...
-        - psutil  # undeclared extra, used preferentially for logging
-        - pycurl  # undeclared extra, used preferentially for performance
         - python >=3.7
         - python-dateutil
         - requests
         - sqlalchemy >=1.1
         - tornado >=5.1
         - traitlets >=4.3.2
+      run_constrained:
+        - psutil >=5.6.5
```
Comment (minrk) [inline recipe/meta.yaml:57]: Where did this version constraint come from?

**Review by minrk — APPROVED** [review]
Looks AOK to me - only one question about where the psutil version constraint came from, since I didn't see it before and I don't recall JupyterHub being sensitive to it, but all looks good to me. Thanks!

**Comment by bollwyvl** [comment]
Couple comments up, also:

https://github.com/jupyterhub/jupyterhub/blob/f6eec29aa222276ea8a9d8066eabb53e9f65b1a7/requirements.txt#L11

**Comment by minrk** [comment]
Got it, thanks!

**Pull request merged**